src/r2b2/minerva.py

```diff
@@ -0,0 +1,218 @@
+        """
+
+        for possible_kmin in range(self.rounds[-1] // 2 + 1, len(self.distribution_null)):
+            tail_null = sum(self.distribution_null[possible_kmin:])
+            tail_reported = sum(self.distribution_reported_tally[possible_kmin:])
+
+            # Minerva's stopping criterion: tail_reported / tail_null > 1 / alpha.
+            if self.alpha * tail_reported > tail_null:
+                if append:
+                    self.min_winner_ballots.append(possible_kmin)
+                return possible_kmin
+
+        # Sentinel of None plays nice with truncation.
+        if append:
+            self.min_winner_ballots.append(None)
+        return None
+
+    def compute_all_min_winner_ballots(self, max_sample_size: int = None, *args, **kwargs):
+        """Compute the minimum number of winner ballots for the complete (that is, ballot-by-ballot)
+        round schedule. Note that Minerva ballot-by-ballot is equivalent to the BRAVO audit.
+
+        Note: Due to limited convolutional precision, results may be off somewhat after the
+        stopping probability very nearly equals 1.
+
+        Args:
+            max_sample_size (int): Optionally set the maximum sample size to generate stopping sizes
+            (kmins) up to. If not provided the maximum sample size is determined by max_frac_to_draw
+            and the total contest ballots.
+
+        Returns:
+            None, kmins are appended to the min_winner_ballots list.
+        """
+
+        if len(self.rounds) > 0:
+            raise Exception("This audit already has an (at least partial) round schedule.")
+        if max_sample_size is None:
+            max_sample_size = math.ceil(self.contest.contest_ballots * self.max_fraction_to_draw)
+        if max_sample_size < self.min_sample_size:
+            raise ValueError("Maximum sample size must be greater than or equal to minimum size.")
+        if max_sample_size > self.contest.contest_ballots:
+            raise ValueError("Maximum sample size cannot exceed total contest ballots.")
+
+        for sample_size in range(self.min_sample_size, max_sample_size + 1):
+            self.rounds.append(sample_size)
+            self.current_dist_null()
+            self.current_dist_reported()
+            # First kmin computed directly.
+            if sample_size == self.min_sample_size:
+                current_kmin = self.find_kmin(True)
+            else:
+                tail_null = sum(self.distribution_null[current_kmin:])
+                tail_reported = sum(self.distribution_reported_tally[current_kmin:])
+                if self.alpha * tail_reported > tail_null:
+                    self.min_winner_ballots.append(current_kmin)
+                else:
+                    current_kmin += 1
+                    self.min_winner_ballots.append(current_kmin)
+            self.truncate_dist_null()
+            self.truncate_dist_reported()
+
+    def compute_risk(self, *args, **kwargs):
+        pass
```

Comment:
What do you think of implementing this as a 'theoretical round' type computation? 

After a round size has been appended and the distributions updated, then someone could run `compute_risk(example_winner_ballots)` with a potential count of winner ballots for that round and get the p_value without modifying the audit state. Basically do exactly what happens in `stopping_condition()` but without updating the p_value schedule and then returning the p_value.

An implementation like that would be very helpful for the simulations where we want to record/analyze p-values, but its also easy enough to implement just within the simulation class if you think it doesn't belong in the Minerva class.

Reply:
Yes, this is a great idea!